### ISSUE TITLE
How can I obtain the -43.2 dB GRU PA model from the benchmark?

### ISSUE THREAD

**Issue by chenjuan456** [lab-emi/OpenDPD#15]
First of all, thank you very much to the OpenDPD team for developing such an excellent open-source platform! OpenDPD provides the neural network DPD research community with valuable datasets, end-to-end learning frameworks, and a wide range of backbone models for both PA and DPD, significantly lowering the barrier to entry for research.

While reading the OpenDPD benchmark report, I noticed that the APA_200MHz dataset uses a GRU PA model, achieving an awesome NMSE performance of -43.52 dB. However, when I trained a GRU PA model with hidden size 23 on the APA_200MHz dataset, I could only achieve an NMSE of approximately -33 dB. I am wondering how I can achieve a PA model with performance under -40 dB, since the PA model's performance is especially critical in the end-to-end learning framework.

I trained the PA using the following command:

```bash
python main.py --step train_pa --dataset_name APA_200MHz --PA_backbone gru --PA_hidden_size 23 --n_epochs 300 --opt_type adamw --frame_length 100/200/300
```
The corresponding training logs are attached below:
[PA_S_0_M_GRU_H_23_F_100_P_1911.csv](https://github.com/user-attachments/files/26749502/PA_S_0_M_GRU_H_23_F_100_P_1911.csv)

[PA_S_0_M_GRU_H_23_F_200_P_1911.csv](https://github.com/user-attachments/files/26749507/PA_S_0_M_GRU_H_23_F_200_P_1911.csv)

[PA_S_0_M_GRU_H_23_F_300_P_1911.csv](https://github.com/user-attachments/files/26749510/PA_S_0_M_GRU_H_23_F_300_P_1911.csv)

Could this performance gap be related to my specific configuration or training setup? Any advice would be greatly appreciated.

**Comment by gaochangw** [comment]
Hi @chenjuan456,

Thank you very much for pointing this out and for sharing your command and training logs. They helped us identify an error in the benchmark report.

After auditing and rerunning the benchmark, we could not reproduce the reported `-43.52 dB` GRU result. That number was incorrect and was not associated with a valid downloadable checkpoint. We apologize for the confusion.

With the corrected benchmark, the APA_200MHz GRU-H28 PA model achieves:

- Validation NMSE: `-38.8504 dB`
- Test NMSE: `-38.9367 dB`
- Selected epoch: `292`

Your approximately `-33 dB` result is consistent with the earlier default training configuration. In particular, recurrent PA models are sensitive to the initial learning rate, batch size, and learning-rate schedule. The corrected benchmark uses:

```bash
python main.py \
  --step train_pa \
  --dataset_name APA_200MHz \
  --PA_backbone gru \
  --PA_hidden_size 28 \
  --PA_num_layers 1 \
  --frame_length 200 \
  --frame_stride 1 \
  --n_epochs 300 \
  --opt_type adamw \
  --lr 5e-3 \
  --lr_schedule 1 \
  --lr_end 5e-5 \
  --decay_factor 0.5 \
  --patience 5 \
  --batch_size 64 \
  --batch_size_eval 64 \
  --loss_type l2 \
  --grad_clip_val 200 \
  --seed 0 \
  --re_level soft \
  --accelerator cuda \
  --devices 0
```

The resulting checkpoint is written to:

```text
save/APA_200MHz/train_pa/PA_S_0_M_GRU_H_28_F_200_P_2746.pt
```

Small numerical differences are possible because the benchmark uses soft CUDA reproducibility.

The updated benchmark does not claim a sub-`-40 dB` GRU result on APA_200MHz; the independently evaluated PA backbones converge around the dataset's approximately `-39 dB` empirical modeling floor. The complete corrected benchmark can also be reproduced with:

```bash
bash benchmark/reproduce_benchmark_report.sh --device 0
```

Thank you again for catching this inconsistency and bringing it to our attention. We have corrected the report and made the training recipe and result provenance explicit.

**Comment by chenjuan456** [comment]
Thanks for the detailed explanation and the corrected config. I've verified that the new configuration works.😊
